cpu/nrf51822/radio/nrfmin/nrfmin.c

```diff
@@ -0,0 +1,670 @@
+    NRF_RADIO->PREFIX0 = CONF_ADDR_PREFIX0;
+    NRF_RADIO->BASE0   = (NRFMIN_DEFAULT_PAN << 16) | _addr;
+    NRF_RADIO->BASE1   = (NRFMIN_DEFAULT_PAN << 16) | CONF_ADDR_BCAST;
+    NRF_RADIO->TXADDRESS = 0x00UL;      /* always send from address 0 */
+    NRF_RADIO->RXADDRESSES = 0x03UL;    /* listen to addresses 0 and 1 */
+    /* configure data fields and packet length whitening and endianess */
+    NRF_RADIO->PCNF0 = (CONF_S1 << RADIO_PCNF0_S1LEN_Pos) |
+                       (CONF_S0 << RADIO_PCNF0_S0LEN_Pos) |
+                       (CONF_LEN << RADIO_PCNF0_LFLEN_Pos);
+    NRF_RADIO->PCNF1 = (CONF_WHITENING << RADIO_PCNF1_WHITEEN_Pos) |
+                       (CONF_ENDIAN << RADIO_PCNF1_ENDIAN_Pos) |
+                       (CONF_BASE_ADDR_LEN << RADIO_PCNF1_BALEN_Pos) |
+                       (CONF_STATLEN << RADIO_PCNF1_STATLEN_Pos) |
+                       (CONF_PAYLOAD_LEN << RADIO_PCNF1_MAXLEN_Pos);
+    /* configure CRC unit */
+    NRF_RADIO->CRCCNF = CONF_CRC_LEN;
+    NRF_RADIO->CRCPOLY = CONF_CRC_POLY;
+    NRF_RADIO->CRCINIT = CONF_CRC_INIT;
+    /* set shortcuts for more efficient transfer */
+    NRF_RADIO->SHORTS = (1 << RADIO_SHORTS_READY_START_Pos);
+    /* enable interrupts */
+    NVIC_SetPriority(RADIO_IRQn, RADIO_IRQ_PRIO);
+    NVIC_EnableIRQ(RADIO_IRQn);
+    /* enable END interrupt */
+    NRF_RADIO->EVENTS_END = 0;
+    NRF_RADIO->INTENSET = (1 << RADIO_INTENSET_END_Pos);
+    /* put device in receive mode */
+    _switch_to_rx();
+    return 0;
+}
+
+int _send(ng_netdev_t *dev, ng_pktsnip_t *pkt)
+{
+    (void)dev;
+    size_t size;
+    size_t pos = 0;
+    uint8_t *dst_addr;
+    ng_netif_hdr_t *hdr;
+    ng_pktsnip_t *payload;
+
+    /* check packet */
+    if (pkt == NULL || pkt->next == NULL) {
+        DEBUG("nrfmin: Error sending packet: packet incomplete\n");
+        return -ENOMSG;
+    }
+
+    /* check if payload is withing length bounds */
+    size = ng_pkt_len(pkt->next);
+    if (size > CONF_PAYLOAD_LEN) {
+        ng_pktbuf_release(pkt);
+        DEBUG("nrfmin: Error sending packet: payload to large\n");
+        return -EOVERFLOW;
+    }
+    /* get netif header and check address length */
+    hdr = (ng_netif_hdr_t *)pkt->data;
+    if (hdr->dst_l2addr_len != 2) {
+        DEBUG("nrfmin: Error sending packet: dest address has invalid size\n");
+        ng_pktbuf_release(pkt);
+        return -ENOMSG;
+    }
+    dst_addr = ng_netif_hdr_get_dst_addr(hdr);
+
+    DEBUG("nrfmin: Sending packet to %02x:%02x - size %u\n",
+           dst_addr[0], dst_addr[1], size);
+
+    /* wait for any ongoing transmission to finish */
+    while (_state == STATE_TX);
+    /* write data into TX buffer */
+    payload = pkt->next;
+    _tx_buf.length = 6 + size;
+    _tx_buf.src_addr[0] = (uint8_t)(_addr >> 8);
+    _tx_buf.src_addr[1] = (uint8_t)(_addr);
+    _tx_buf.dst_addr[0] = dst_addr[0];
+    _tx_buf.dst_addr[1] = dst_addr[1];
+    _tx_buf.proto = (uint16_t)payload->type;
```

Comment:
Since this the values of `nettype_t` are dependent on the module make-up during compile time I would highly recommend to use the ethertype (also 16-bit long) or similar for `proto`. Otherwise you are basically putting the constraint on this driver that it only works with an equally compiled network stack in RIOT.